TLSharp.Core/Network/TcpTransport.cs

```diff
@@ -11,12 +12,18 @@ public class TcpTransport : IDisposable
         private readonly TcpClient _tcpClient;
         private int sendCounter = 0;
 
-        public TcpTransport(string address, int port)
+        public TcpTransport(string address, int port,
+            string httpProxyHost = "", int httpProxyPort = 0, string proxyUserName = "", string proxyPassword = "")
```

Comment:
Instead of passing 4 parameters, can you wrap these elements into a class (e.g. ProxyInfo?) that contains two properties: Uri (this way it would include host and port) and NetworkCredentials (which includes user and password).